Integrate django-ansible-base

jessicamack opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
Test using django-ansible-base in AWX
<!---
If you are fixing an existing issue, please include "related #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - New or Enhanced Feature

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
 - API

##### AWX VERSION
<!--- Paste verbatim output from `make VERSION` between quotes below -->
```

```


##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```

```

jessicamack committed:
add django-ansible-base to requirements

Signed-off-by: jessicamack <jmack@redhat.com>

jessicamack committed:
add django-ansible-base to requirements

Signed-off-by: jessicamack <jmack@redhat.com>

jessicamack committed:
add django-ansible-base to requirements

Signed-off-by: jessicamack <jmack@redhat.com>

jessicamack committed:
add django-ansible-base to requirements

Signed-off-by: jessicamack <jmack@redhat.com>

jessicamack committed:
add django-ansible-base to requirements

Signed-off-by: jessicamack <jmack@redhat.com>

john-westcott-iv commented:
This is awesome for me to see this work started. I see this is still in Draft so maybe you are already planning on all of the following but just so its documented in this issue:

This will add django-ansible-base to AWX itself but we really want to determine the feasibility of replacing the SSO application (awx/sso) with the authentictors code in django-ansible-base. See the documentation at https://github.com/ansible/django-ansible-base/blob/devel/docs/authentication.md for more details about how to enable django-ansible-base authentication. Probably the biggest question that we want to answer from the experiment is what happens if you remove the migrations for awx/sso and then attempt to apply migrations for django-ansible-base? Will this cause a large rollback in any of the other applications or does nothing depend on awx/sso? We may also need additional migration scripts to help map users' fields from awx/sso to users in django-ansible-base. To test this you likely will want to setup the LDAP/Keycloak side cars with AWX and create some external accounts (of each type) and then try the migrations to see what data is "lost" from removing awx/sso migrations. 

Let me know if you want to meet to discuss any of this or have any questions.

jessicamack committed:
Merge remote-tracking branch 'upstream/devel' into feature_django-ansible-base

jessicamack committed:
fix requirement and add license

jessicamack committed:
fix requirement and add license

jessicamack committed:
fix requirement and add license

jessicamack committed:
fix requirement and add license

jessicamack committed:
fix requirement and add license

lucas-benedito committed:
Setting credential_type as required (#14651)

* Setting credential_type as required

* Added test for missing credential_type in credential module

* Corrected test assertion

---------

Co-authored-by: Lucas Benedito <lbenedit@redhat.com>

tvo318 committed:
Replaced with larger graphic. (#14647)

AlanCoding committed:
Correctly handle case where unpartitioned table does not exist (#14648)

unnecessary-username committed:
Set subscription type as developer for developer subscriptions. (#14584)

* Set subscription type as developer for developer subscriptions.

Signed-off-by: Tong He <the@redhat.com>

* Set subscription type as developer for developer subscription manifests.

Signed-off-by: Tong He <the@redhat.com>

* Remedy the wrong character to assign value.

Signed-off-by: Tong He <the@redhat.com>

* Reformat licensing.py by black.

Signed-off-by: Tong He <the@redhat.com>

---------

Signed-off-by: Tong He <the@redhat.com>

oraNod committed:
issue #14653 heading does not render correctly (#14665)

sscheib-rh committed:
Adding the possibility to decode base64 decoded strings to Delinea's Devops Secret Vault (DSV) (#14646)

Adding the possibility to decode base64 decoded strings to Delinea's Devops Secret Vault (DSV).
This is necessary as uploading files to DSV is not possible (and not meant to be) and files should be added base64 encoded.
The commit is making sure to remain backward compatible (no secret decoding), as a default is supplied.

This has been tested with DSV and works for secrets that are base64 encoded and secrets that are not base64 encoded (which is the default).

Signed-off-by: Steffen Scheib <sscheib@redhat.com>

tvo318 committed:
Added missing pointers to release notes (#14659)

* Replaced with larger graphic.

* Revert "Replaced with larger graphic."

This reverts commit 1214b00052b43c46c5ee9b2833e61c779884ec1c.

* Added missing pointers to release notes.

TheRealHaoLiu committed:
Make vault init more idempotent (#14664)

Currently if you cleanup docker volume for vault and bring docker-compose development back up with vault enabled we will not initialize vault because the secret files still exist.

This change will attempt to initialize vault reguardless and update the secret file if vault is initialized

AlanCoding committed:
Remove unused methods we attach to user model (#14668)

AlanCoding committed:
Update RBAC docs, remove unused get_permissions (#14492)

* Update RBAC docs, remove unused get_permissions

* Add back in section for get_roles_on_resource

oraNod committed:
Upgrade doc requirements (#14669)

* upgrade when pip compiling doc reqs

* upgrade doc requirements

chrismeyersfsu committed:
allow pytest --migrations to succeed (#14663)

* allow pytest --migrations to succeed

* We actually subvert migrations from running in test via pytest.ini
  --no-migrations option. This has led to bit rot for the sqlite
  migrations happy path. This changeset pays off that tech debt and
  allows for an sqlite migration happy path.
* This paves the way for programatic invocation of individual migrations
  and weaving of the creation of resources (i.e. Instance, Job Template,
  etc). With this, a developer can instantiate various database states,
  trigger a migration, assert the state of the db, and then have pytest
  rollback all of that.
* I will note that in practice, running these migrations is dog shit
  slow BUT this work also opens up the possibility of saving and
  re-using sqlite3 database files. Normally, caching is not THE answer
  and causes more harm than good. But in this case, our migrations are
  mostly write-once (I say mostly because this change set violates
  that :) so cache invalidation isn't a major issue.

* functional test for migrations on sqlite

* We commonly subvert running migrations in test land. Test land uses
  sqlite. By not constantly exercising this code path it atrophies. The
  smoke test here is to continuously exercise that code path.
* Add ci test to run migration tests separately, they take =~ 2-3
  minutes each on my laptop.
* The smoke tests also serves as an example of how to write migration
  tests.

* run migration tests in ci

jessicamack committed:
upgrade requirements

jessicamack committed:
upgrade requirements

jessicamack committed:
update settings and urls

jessicamack committed:
update settings and urls

jessicamack committed:
update settings and urls

jessicamack committed:
Merge branch 'feature_django-ansible-base' of https://github.com/jessicamack/awx into feature_django-ansible-base

jessicamack committed:
add back previous social auth config

jessicamack committed:
add back previous social auth config

jessicamack committed:
add back previous social auth config

shanemcd commented on awx/settings/defaults.py:
```diff
@@ -370,12 +372,7 @@
-    'awx.sso.backends.LDAPBackend3',
-    'awx.sso.backends.LDAPBackend4',
-    'awx.sso.backends.LDAPBackend5',
+    "ansible_base.authentication.backend.AnsibleBaseAuth",
```
Are we actually ready to make this change? Does this "just work", or will it require some additional changes (especially on the UI)?

jessicamack replied:
It does "just work" but I've been using the DRF UI. I believe the AWX UI work is not complete yet. @john-westcott-iv can you confirm this?

AlanCoding commented on awx/urls.py:
```diff
@@ -5,12 +5,14 @@
     re_path(r'', include('awx.ui.urls', namespace='ui')),
     re_path(r'^ui_next/.*', include('awx.ui_next.urls', namespace='ui_next')),
     re_path(r'^api/', include('awx.api.urls', namespace='api')),
+    re_path(r'^api/v1/', include(base_urls)),
```
I want to draw some broader attention to this. My gut reaction is that it can't possibly be a good idea to add endpoints to `/api/v1/`, because that used to exist in a prior version, and it was deprecated and then removed.

jessicamack replied:
Good point. I was following the docs so used v1 but that wouldn't fit within the context of AWX.

gundalow reviewed (changes requested):
1. Let's update this PR title & description to only update the requirements
2. revert the changes to `defaults.py` and `urls.py`.
3. Rebase
4. Merge once CI is green

jessicamack committed:
Revert "update settings and urls"

This reverts commit ef6d2dd242c6ef0cf452811098b187f790f035c4.

jessicamack committed:
Revert "update settings and urls"

This reverts commit ef6d2dd242c6ef0cf452811098b187f790f035c4.

jessicamack committed:
Revert "update settings and urls"

This reverts commit ef6d2dd242c6ef0cf452811098b187f790f035c4.

chrismeyersfsu commented on requirements/requirements.txt:
```diff
@@ -406,9 +407,9 @@ slack-sdk==3.19.4
 smmap==5.0.0
     # via gitdb
-social-auth-app-django==5.0.0
+social-auth-app-django==5.4.0
```
When I see a .0 z stream (semantic versioning `x.y.z`) I always wonder "Is there a newer .1. or .2 version available?". With semantic versioning a new `y` stream indicates new features or breaking changes. With new features comes new bugs. I guess I'm suspicious of new bugs with `.0` releases.

HOWEVER, in this case if we go to `social-auth-app-django` releases https://github.com/python-social-auth/social-app-django/releases they only have `y` releases.

I guess I'm still suspicious of this projects maturity BUT if I zoom out, this project looks like it's sort of a wrapper to djangoize another library. So I guess the lack of bug fix releases is justified.

Not asking for any action here, just brain dumping how I consider dependency updates.

AlanCoding replied:
I assume this comes from ansible_base

https://github.com/ansible/django-ansible-base/blob/devel/requirements/requirements.txt#L100

Now, the pyproject.toml references requirements.in file, not that. So I don't believe this work strictly requires upgrading to get it in sync. Especially if we're not using the related views.

jessicamack replied:
It does come from ansible_base. @AlanCoding before upgrading the CI was failing on a check_migrations call because the models use a change that happens in the upgraded version of social_auth_app_django. That hanging migration might not affect functionality for what you need though.

AlanCoding replied:
That makes sense to me, I would prefer to just go with the newer library version then.

chrismeyersfsu commented on requirements/requirements.txt:
```diff
@@ -406,9 +407,9 @@ slack-sdk==3.19.4
+social-auth-app-django==5.4.0
     # via -r /awx_devel/requirements/requirements.in
-social-auth-core[openidconnect]==4.3.0
+social-auth-core[openidconnect]==4.4.1
```
Looks like there is a 4.4.2 release. Can we pull that one in instead?

https://github.com/python-social-auth/social-core/releases

jessicamack replied:
Yes, version has been changed to 4.4.2

jessicamack committed:
Merge branch 'feature_django-ansible-base' of https://github.com/jessicamack/awx into feature_django-ansible-base

jessicamack committed:
temp remove package

Signed-off-by: jessicamack <jmack@redhat.com>

jessicamack committed:
temp remove package

Signed-off-by: jessicamack <jmack@redhat.com>

jessicamack committed:
add back package

Signed-off-by: jessicamack <jmack@redhat.com>

jessicamack committed:
Merge branch 'feature_django-ansible-base' of https://github.com/jessicamack/awx into feature_django-ansible-base

jessicamack commented:
> 1. Let's update this PR title & description to only update the requirements
> 2. revert the changes to `defaults.py` and `urls.py`.
> 3. Rebase
> 4. Merge once CI is green

Rebase hit a conflict that introduced unrelated commits to the PR. In the interest of getting this merged faster I opened a clean PR https://github.com/ansible/awx/pull/14705 with just the discussed changes. Converting this back to a draft.
